File: languages/java/app-encryption/src/main/java/com/godaddy/asherah/appencryption/envelope/EnvelopeEncryptionBytesImpl.java
package com.godaddy.asherah.appencryption.envelope;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import com.godaddy.asherah.appencryption.utils.Json;

public class EnvelopeEncryptionBytesImpl implements EnvelopeEncryption<byte[]> {
  private static final Logger logger = LoggerFactory.getLogger(EnvelopeEncryptionBytesImpl.class);

  private final EnvelopeEncryptionJsonImpl envelopeEncryptionJsonImpl;

  public EnvelopeEncryptionBytesImpl(final EnvelopeEncryptionJsonImpl envelopeEncryptionJsonImpl) {
    this.envelopeEncryptionJsonImpl = envelopeEncryptionJsonImpl;
  }

  @Override
  public byte[] decryptDataRowRecord(final byte[] dataRowRecord) {
    Json dataRowRecordJson = new Json(dataRowRecord);
    return envelopeEncryptionJsonImpl.decryptDataRowRecord(dataRowRecordJson.toJsonObject());
  }

  @Override
  public byte[] encryptPayload(final byte[] payload) {
    Json drrJson = new Json(envelopeEncryptionJsonImpl.encryptPayload(payload));
    return drrJson.toUtf8();
  }

  @Override
  public void close() {
    try {
      envelopeEncryptionJsonImpl.close();
    }
    catch (Exception e) {
      logger.error("unexpected exception during close", e);
    }
  }

}

File: languages/csharp/AppEncryption/AppEncryption/Envelope/EnvelopeEncryptionBytesImpl.cs
using System;
using GoDaddy.Asherah.AppEncryption.Util;
using GoDaddy.Asherah.Logging;
using Microsoft.Extensions.Logging;
using Newtonsoft.Json.Linq;

namespace GoDaddy.Asherah.AppEncryption.Envelope
{
    public class EnvelopeEncryptionBytesImpl : IEnvelopeEncryption<byte[]>
    {
        private static readonly ILogger Logger = LogManager.CreateLogger<EnvelopeEncryptionBytesImpl>();

        private readonly IEnvelopeEncryption<JObject> envelopeEncryptionJson;

        public EnvelopeEncryptionBytesImpl(IEnvelopeEncryption<JObject> envelopeEncryptionJson)
        {
            this.envelopeEncryptionJson = envelopeEncryptionJson;
        }

        public void Dispose()
        {
            try
            {
                envelopeEncryptionJson.Dispose();
            }
            catch (Exception e)
            {
                Logger.LogError(e, "Unexpected exception during dispose");
            }
        }

        public virtual byte[] DecryptDataRowRecord(byte[] dataRowRecord)
        {
            Json dataRowRecordJson = new Json(dataRowRecord);
            return envelopeEncryptionJson.DecryptDataRowRecord(dataRowRecordJson.ToJObject());
        }

        public virtual byte[] EncryptPayload(byte[] payload)
        {
            Json drrJson = new Json(envelopeEncryptionJson.EncryptPayload(payload));
            return drrJson.ToUtf8();
        }
    }
}
